### DISCUSSION TITLE
`nvdaControllerClient.dll` location?

### DISCUSSION THREAD

**Discussion by C-Loftus** [nvaccess/nvda#15819] [Q&A] [answered]
I'm a developer and am trying to use nvda to speak text output from another program.  I wanted to use the controller client as shown in https://github.com/nvaccess/nvda/blob/master/extras/controllerClient/x86/example_python.py

However I cannot find the necessary dll file `nvdaControllerClient.dll`. It is not within my C:\Program Files (x86)\NVDA folder. Is there somewhere else I should be searching for this? I cannot find any other information online about this.  Has this been deprecated?

**Comment by C-Loftus** [comment, marked as answer]
Never mind, I did not realize it needed to be downloaded according to the info here `https://github.com/nvaccess/nvda/blob/master/extras/controllerClient/readme.md`

**Reply by armughal189** [reply]
Can I use this library outside Python like in My C# or C++ Apps as well?
And, Can I have to do some extra setups to use it in C#?

**Reply by hwf1324** [reply]
> Can I use this library outside Python like in My C# or C++ Apps as well?
> And, Can I have to do some extra setups to use it in C#?

https://github.com/nvaccess/nvda/tree/master/extras%2FcontrollerClient%2Fexamples%2Fexample_csharp
